pkg/plugins/manager/loader/loader.go

```diff
@@ -72,17 +83,48 @@ func (l *Loader) Load(ctx context.Context, src plugins.PluginSource) ([]*plugins
 		}
 		bootstrappedPlugins = append(bootstrappedPlugins, bootstrappedPlugin...)
 	}
+	l.log.Debug("Bootstrapped", "class", src.PluginClass(ctx), "duration", time.Since(st))
 
+	st = time.Now()
 	validatedPlugins := []*plugins.Plugin{}
+	type validateResult struct {
+		bootstrappedPlugin *plugins.Plugin
+		err                error
+	}
+	validateResults := make(chan validateResult, len(bootstrappedPlugins))
+
+	// If the PluginsCDNSyncLoaderEnabled feature is enabled, validate plugins in parallel.
+	// Otherwise, validate plugins sequentially.
+	var limitSize int
+	if l.cfg.Features.PluginsCDNSyncLoaderEnabled && src.PluginClass(ctx) == plugins.ClassCDN {
+		limitSize = min(len(bootstrappedPlugins), concurrencyLimit)
+	} else {
+		limitSize = 1
+	}
+	limit := make(chan struct{}, limitSize)
 	for _, bootstrappedPlugin := range bootstrappedPlugins {
-		err := l.validation.Validate(ctx, bootstrappedPlugin)
-		if err != nil {
-			l.recordError(ctx, bootstrappedPlugin, err)
+		bootstrappedPlugin := bootstrappedPlugin
+		limit <- struct{}{}
+		go func() {
+			err := l.validation.Validate(ctx, bootstrappedPlugin)
+			validateResults <- validateResult{
+				bootstrappedPlugin: bootstrappedPlugin,
+				err:                err,
+			}
+			<-limit
+		}()
+	}
+	for i := 0; i < len(bootstrappedPlugins); i++ {
+		r := <-validateResults
+		if r.err != nil {
+			l.recordError(ctx, r.bootstrappedPlugin, r.err)
 			continue
 		}
-		validatedPlugins = append(validatedPlugins, bootstrappedPlugin)
+		validatedPlugins = append(validatedPlugins, r.bootstrappedPlugin)
 	}
+	l.log.Debug("Validated", "class", src.PluginClass(ctx), "duration", time.Since(st))
```

Comment:
```suggestion
	l.log.Debug("Validated", "class", src.PluginClass(ctx), "duration", time.Since(st), "total", len(validatedPlugins))
```

This could also be useful when comparing durations in the wild maybe?

Reply:
:+1: added